google-cloud-storage/src/test/java/com/google/cloud/storage/StorageImplMockitoTest.java

```diff
@@ -442,4 +448,75 @@ public void testFailedBatchWithRetry() {
     }
     verify(batchMock, times(6)).submit();
   }
+
+  @Test
+  public void testBatchRetry() {
+    final BlobId blobId1 = BlobId.of(BUCKET_NAME1, BLOB_NAME1);
+    final BlobId blobId2 = BlobId.of(BUCKET_NAME2, BLOB_NAME2);
+    options =
+        StorageOptions.newBuilder()
+            .setProjectId("projectId")
+            .setClock(TIME_SOURCE)
+            .setServiceRpcFactory(rpcFactoryMock)
+            .setRetrySettings(
+                ServiceOptions.getDefaultRetrySettings().toBuilder().setMaxAttempts(5).build())
+            .build();
+    RpcBatch batchMock = mock(RpcBatch.class);
+    when(storageRpcMock.createBatch()).thenReturn(batchMock);
+    doAnswer(
+            new Answer<Void>() {
+              @Override
+              public Void answer(InvocationOnMock invocation) throws Throwable {
+                RpcBatch.Callback<StorageObject> callback = invocation.getArgument(1);
+                callback.onSuccess(blobId1.toPb());
+                return null;
+              }
+            })
+        .when(batchMock)
+        .addGet(
+            Mockito.eq(blobId1.toPb()),
+            Mockito.<RpcBatch.Callback<StorageObject>>any(),
+            Mockito.<StorageRpc.Option, Object>anyMap());
+    doAnswer(
+            new Answer<Void>() {
+              @Override
+              public Void answer(InvocationOnMock invocation) throws Throwable {
+                RpcBatch.Callback<StorageObject> callback = invocation.getArgument(1);
+                GOOGLE_JSON_ERROR.setCode(503);
+                GOOGLE_JSON_ERROR.setMessage("message");
+                callback.onFailure(GOOGLE_JSON_ERROR);
+                return null;
+              }
+            })
+        .when(batchMock)
+        .addGet(
+            Mockito.eq(blobId2.toPb()),
+            Mockito.<RpcBatch.Callback<StorageObject>>any(),
+            Mockito.<StorageRpc.Option, Object>anyMap());
+    initializeService();
+    StorageBatch batch = storage.batch();
+    doThrow(new StorageException(500, "batchError"))
+        .doThrow(new StorageException(500, "batchError"))
+        .doNothing()
+        .when(batchMock)
+        .submit();
+    StorageBatchResult<Blob> blobStorageBatchResult1 = batch.get(blobId1);
+    StorageBatchResult<Blob> blobStorageBatchResult2 = batch.get(blobId2);
+    batch.submit();
+    Blob actualBlob = blobStorageBatchResult1.get();
+    assertEquals(expectedBlob1.getName(), actualBlob.getName());
+    assertEquals(expectedBlob1.getBucket(), actualBlob.getBucket());
+    try {
+      blobStorageBatchResult2.get();
+      fail("");
+    } catch (StorageException ex) {
+      assertEquals("message", ex.getMessage());
+    }
```

Comment:
please use variables instead of 'message'
please add assertEquals for ex.getCode()

Reply:
Done.